packetery/src/Packetery/Module/Upgrade.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Class Upgrade.
+ *
+ * @package Packetery\Module
+ */
+
+declare( strict_types=1 );
+
+namespace Packetery\Module;
+
+use Packetery\Module\Order\DbRepository;
+use Packetery\Module\Order\Entity;
+
+/**
+ * Class Upgrade.
+ */
+class Upgrade {
+
+	/**
+	 * Order repository.
+	 *
+	 * @var DbRepository
+	 */
+	private $orderRepository;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param DbRepository $orderRepository Order repository.
+	 */
+	public function __construct( DbRepository $orderRepository ) {
+		$this->orderRepository = $orderRepository;
+	}
+
+	/**
+	 * Checks previous plugin version and runs upgrade if needed.
+	 * https://www.sitepoint.com/wordpress-plugin-updates-right-way/
+	 *
+	 * @return void
+	 */
+	public function check(): void {
+		$oldVersion = get_option( 'packetery_plugin_version' );
+		if ( Plugin::VERSION !== $oldVersion ) {
+			// Update the version if no upgrade is run.
+			$result = true;
+
+			// First version which is saved. We will add version_compare later.
+			if ( ! $oldVersion ) {
+				$result = $this->upgrade_1_1_2();
+			}
+
+			if ( $result ) {
+				update_option( 'packetery_plugin_version', Plugin::VERSION );
```

Comment:
aktualizaci packetery_plugin_version bych dělal vždy. $result v tomto případě nikdy není false.

Reply:
Je to informace o nainstalované verzi. Pokud upgrade selže, jde verzi považovat za nainstalovanou?

Reply:
Upgrade selhat nesmí. Plugin změny potřebuje. 

Pokud migrace způsobí 500, tak bude mít eshopista jak plugin deaktivovat? Volá se ten event "plugins_loaded" pokaždé?